From: Gautam Menghani <gautammenghani201@gmail.com>
Subject: mm-khugepaged-add-tracepoint-to-collapse_file-v4
Date: Sat, 3 Dec 2022 01:48:07 +0530

swap is_shmem and addr to save space, per Steven Rostedt

Link: https://lkml.kernel.org/r/20221202201807.182829-1-gautammenghani201@gmail.com
Signed-off-by: Gautam Menghani <gautammenghani201@gmail.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Masami Hiramatsu (Google) <mhiramat@kernel.org>
Cc: Steven Rostedt (Google) <rostedt@goodmis.org>
Cc: Yang Shi <shy828301@gmail.com>
Cc: Zach O'Keefe <zokeefe@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/trace/events/huge_memory.h |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/include/trace/events/huge_memory.h~mm-khugepaged-add-tracepoint-to-collapse_file-v4
+++ a/include/trace/events/huge_memory.h
@@ -207,13 +207,13 @@ TRACE_EVENT(mm_khugepaged_collapse_file,
 	TP_PROTO(struct mm_struct *mm, struct page *hpage, pgoff_t index,
 			bool is_shmem, unsigned long addr, struct file *file,
 			int nr, int result),
-	TP_ARGS(mm, hpage, index, is_shmem, addr, file, nr, result),
+	TP_ARGS(mm, hpage, index, addr, is_shmem, file, nr, result),
 	TP_STRUCT__entry(
 		__field(struct mm_struct *, mm)
 		__field(unsigned long, hpfn)
 		__field(pgoff_t, index)
-		__field(bool, is_shmem)
 		__field(unsigned long, addr)
+		__field(bool, is_shmem)
 		__string(filename, file->f_path.dentry->d_iname)
 		__field(int, nr)
 		__field(int, result)
@@ -223,19 +223,19 @@ TRACE_EVENT(mm_khugepaged_collapse_file,
 		__entry->mm = mm;
 		__entry->hpfn = hpage ? page_to_pfn(hpage) : -1;
 		__entry->index = index;
-		__entry->is_shmem = is_shmem;
 		__entry->addr = addr;
+		__entry->is_shmem = is_shmem;
 		__assign_str(filename, file->f_path.dentry->d_iname);
 		__entry->nr = nr;
 		__entry->result = result;
 	),
 
-	TP_printk("mm=%p, hpage_pfn=0x%lx, index=%ld, is_shmem=%d, addr=%ld, filename=%s, nr=%d, result=%s",
+	TP_printk("mm=%p, hpage_pfn=0x%lx, index=%ld, addr=%ld, is_shmem=%d, filename=%s, nr=%d, result=%s",
 		__entry->mm,
 		__entry->hpfn,
 		__entry->index,
-		__entry->is_shmem,
 		__entry->addr,
+		__entry->is_shmem,
 		__get_str(filename),
 		__entry->nr,
 		__print_symbolic(__entry->result, SCAN_STATUS))
_
